test/e2e/system_service_test.go

```diff
@@ -0,0 +1,136 @@
+package integration
+
+import (
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"os/exec"
+	"strconv"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gexec"
+)
+
+var _ = FDescribe("podman system service", func() {
+	podman := "podman"
+	BeforeEach(func() {
+		if b, found := os.LookupEnv("PODMAN_BINARY"); found {
+			podman = b
+		}
+	})
+
+	Describe("verify timeout", func() {
+		It("of 2 seconds", func() {
+			address := url.URL{
+				Scheme: "tcp",
+				Host:   net.JoinHostPort("localhost", randomPort()),
+			}
+
+			cmd := exec.Command(podman, "system", "service", "--time=2", address.String())
+			session, err := Start(cmd, GinkgoWriter, GinkgoWriter)
+			Expect(err).ShouldNot(HaveOccurred())
+			defer session.Kill()
+
+			WaitForService(address)
+
+			session.Wait(5 * time.Second)
+			Eventually(session, 5).Should(Exit(0))
+		})
+	})
+
+	Describe("verify pprof endpoints", func() {
+		It("are available", func() {
+			address := url.URL{
+				Scheme: "tcp",
+				Host:   net.JoinHostPort("localhost", randomPort()),
+			}
+
+			pprofPort := randomPort()
+			cmd := exec.Command(podman, "system", "service", "--time=0",
+				"--pprof-address=localhost:"+pprofPort,
+				address.String())
+			session, err := Start(cmd, GinkgoWriter, GinkgoWriter)
+			Expect(err).ShouldNot(HaveOccurred())
+			defer session.Kill()
+
+			WaitForService(address)
+
+			heap := url.URL{
+				Scheme:   "http",
+				Host:     net.JoinHostPort("localhost", pprofPort),
+				Path:     "/debug/pprof/heap",
+				RawQuery: "seconds=2",
+			}
+			resp, err := http.Get(heap.String())
+			Expect(err).ShouldNot(HaveOccurred())
+			defer resp.Body.Close()
+			Expect(resp).To(HaveHTTPStatus(http.StatusOK))
+
+			body, err := io.ReadAll(resp.Body)
```

Comment:
```suggestion
			body, err := ioutil.ReadAll(resp.Body)
```
We still use go 1.15 in CI.

Reply:
Interesting, Goland is usually good about pointing out issues like that.